procedures/unit-testing-tracing-analytics.ipf

```diff
@@ -0,0 +1,315 @@
+#pragma rtGlobals = 3
+#pragma TextEncoding = "UTF-8"
+#pragma rtFunctionErrors = 1
+#pragma ModuleName = UTF_Tracing_Analytics
+
+#if (IgorVersion() >= 9.00) && Exists("TUFXOP_Version") && (NumberByKey("BUILD", IgorInfo(0)) >= 38812)
+
+/// @name UTF Analytic output modes
+/// @anchor AnalyticModes
+/// @{
+Constant UTF_ANALYTICS_FUNCTIONS = 0x00
+Constant UTF_ANALYTICS_LINES = 0x01
+/// @}
+
+/// @name UTF Analytics sorting modes
+/// @anchor AnalyticSorting
+/// @{
+Constant UTF_ANALYTICS_CALLS = 0x00
+Constant UTF_ANALYTICS_SUM = 0x01
+/// @}
+
+static Structure CollectionResult
+	WAVE/T functions
+	WAVE lines
+	WAVE calls
+	WAVE sums
+	variable count
+EndStructure
+
+static Function GetMaxFuncCount()
+	WAVE/WAVE funcLocations = UTF_Tracing#GetFuncLocations()
+	variable size = DimSize(funcLocations, UTF_ROW)
+
+	Make/FREE=1/N=(size) helper = DimSize(funcLocations[p][%FUNCLIST], UTF_ROW)
+	return WaveMax(helper)
+End
+
+static Function GetMaxProcLineCount()
+	WAVE wv = UTF_Tracing#GetProcSizes()
+
+	return WaveMax(wv)
+End
+
+static Function/WAVE GetTotals()
+	variable i, numWaves
+
+	TUFXOP_GetStorage/N="IUTF_TestRun" storage
+	numWaves = DimSize(storage, UTF_ROW)
+	WAVE/ZZ totals
+	for(i = 0; i < numWaves; i++)
+		WAVE/WAVE/Z entryOuter = storage[i]
+		if(!WaveExists(entryOuter))
+			continue
+		endif
+		WAVE entry = entryOuter[0]
+		if(WaveExists(totals))
+			totals += entry
+		else
+			Duplicate/FREE=1 entry, totals
+		endif
+	endfor
+
+	if(!WaveExists(totals))
+		Make/N=0/FREE=1 totals
+	endif
+
+	return totals
+End
+
+static Function CollectFunctions(WAVE totals, WAVE/T procs, STRUCT CollectionResult &result)
+	variable i, j, startIndex, endIndex
+	variable procCount = DimSize(procs, UTF_ROW)
+	variable funcCount = GetMaxFuncCount()
+	DFREF dfr = GetPackageFolder()
+	WAVE/WAVE funcLocations = UTF_Tracing#GetFuncLocations()
+	variable lbFuncList = FindDimLabel(funcLocations, UTF_COLUMN, "FUNCLIST")
+	variable lbFuncStart = FindDimLabel(funcLocations, UTF_COLUMN, "FUNCSTART")
+
+	Make/FREE=1/N=(procCount, funcCount)/T result.functions
+	Make/FREE=1/N=(procCount, funcCount) result.lines, result.calls, result.sums
+
+	for(i = 0; i < procCount; i++)
+		WAVE/T procFuncNames = funcLocations[i][lbFuncList]
+		WAVE procFuncLines = funcLocations[i][lbFuncStart]
+		funcCount = DimSize(procFuncNames, UTF_ROW)
+		result.count += funcCount
+
+		result.functions[i][0, funcCount - 1] = procFuncNames[q]
```

Comment:
This obviously fails with an RTE for the case funcCount is zero, e.g. for procedure files containing only constants/structures etc.